### PR TITLE
LuceneRecordCursor should only load stored values needed for query #2145

### PR DESCRIPTION
**Pull request by jleach4**
Test coverage in FDBLuceneQueryTest.covering().  This is just a performance optimization to not load all stored values unless query requires them.

### CODE REVIEW

**Review by MMcM — APPROVED** [review]
We could perhaps trim off a tiny bit more by making a `DocumentStoredFieldVisitor` with the target fields directly, but the one made internally probably goes on the stack anyway.

File: fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java [hunk 1]
```diff
@@ -162,6 +165,10 @@ public class LuceneRecordCursor implements BaseCursor<IndexEntry> {
         this.pageSize = pageSize;
         this.executorService = executorService;
         this.storedFields = storedFields;
+        this.storedFieldsToReturn = Sets.newHashSet(LuceneIndexMaintainer.PRIMARY_KEY_FIELD_NAME);
+        if (this.storedFields != null) { // Only return StoredValues required for primary key and to support query.
```
Comment (ohadzeliger) [inline fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java:169]: Trying to understand the role of the `storedFields` argument: Running queries with various `SELECT` fields (or `.*`) resulted in this parameter unchanged: it always contains 13 elements, 11 of which are `null` and the remaining 2 are `msgstate_uid` and `msgstate_flags`.

Comment (ohadzeliger) [reply fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java:169]: So I guess my question is what should be the intended impact of this optimization?

Comment (jleach4) [reply fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java:169]: The optimization will not materialize the stored values that are not-needed based on the query.  We need primary key for each value but if you had a field for example that stored another value, it is not needed to be retrieved for the user unless the query user needs to return it.

**Review by ohadzeliger — APPROVED** [review]

**Pull request merged**